### ISSUE TITLE
Teach mxRefModels how to handle WLS

### ISSUE THREAD

**Issue by mhunter1** [OpenMx/OpenMx#184]
The saturated and independence models are known.  Could be tricky to make sure Chi-squared values are handled properly in subsequent function calls.

**Comment by tbates** [comment]
Just to document current behavior:
```R
m1 = umxRAM("WLS", data = mtcars, type="WLS",
	umxPath(c("wt"), to = "mpg"),
	umxPath("wt", with = "disp"),
	umxPath(var = c("mpg", "wt", "disp"))
)

refModels = mxRefModels(m1, run = TRUE)
```
> Error in .local(.Object, model, distribution) : 
Don't know how to makedefaultdistributionreference models for a model with a MxFitFunctionWLS fit function.

**Comment by mhunter1** [comment]
Here's my proposal:

1.  Throw a friendly error or something similar when `mxRefModels` is called on a WLS model
1.  Maybe just make `mxRefModels` work for WLS.  It shouldn't be that hard.
1.  Compute the independence model fit in the backend.

The independence fit 
  - needs to be passed back as an attribute: `attr(model@fitfunction$result, "IndependenceLikelihood")`
  - can be computed with a single fit function evaluation.

The fit function evaluation for the function `r'Wr = (s-sigma)'W(s-sigma)` can be done by either of the following:
1.  setting all the covariance elements of `sigma` to zero and everything else equal to the corresponding elements of `s`
1.  setting all the non-covariance elements (means, thresholds, variances) of `r` to zero and leaving everything else alone.

**Comment by mhunter1** [comment]
The code in the backend for standardizing is found by

```
$ grep -in 'normalToStdVector' ./src/*.cpp
./src/Compute.cpp:3731:                         normalToStdVector(o1.covMat, o1.meansMat, o1.slopeMat, o1.thresholdMat,
./src/omxExpectation.cpp:349:void normalToStdVector(omxMatrix *cov, omxMatrix *mean, omxMatrix *slope, omxMatrix *thr,
./src/omxExpectation.cpp:439:   normalToStdVector(cov, getComponent("means"), getComponent("slope"), thresholdsMat,
./src/omxWLSFitFunction.cpp:60: normalToStdVector(cov, means, slope, thresholdMat, numOrdinal, thresholds, vec1);
```

This lets you know which arguments are for the covariances vs the means, thresholds, and variances.

**Comment by mcneale** [comment]
Sorry to miss the call today. Might have worked but for my niece breaking her wrist snowboarding.  She’s ok.
Lmk about minutes.

Cheers
Mike

Michael C. Neale, Ph.D.
Professor, Virginia Institute for Psychiatric and Behavioral GeneticsIPBG
Virginia Commonwealth University
Shipping Address: 800 East Leigh St. Suite 1-156, Richmond VA 23219-1534
Mailing Address : Box 980126 MCV Richmond VA 23298-0126
Telephone: (804) 828-3369  Email: michael.neale@vcuhealth.org
________________________________

**Comment by mhunter1** [comment]
For the independence model, it seems that other software report the covariate-inclusive independence model fit.  This matches my intuitive about what is most appropriate.

**Comment by tbates** [comment]
Any chance to get `mxRefModels` working for WLS?

The problem remains as it was in 2019: The user following the OpenMx instructions about getting fit indices is just told:
> Error in .local(.Object, model, distribution) : 
  Don't know how to make 'default' distribution reference models for a model with a MxFitFunctionWLS fit function.

Without this functionality, key fit indices for WLS models can't be had, which renders WLS in OpenMx non-useful for publishable work.

**Comment by mcneale** [comment]
Do you have the statistical knowledge necessary to implement this?  Where is the math?  Such comparisons may be left out due to uncertainty as to whether models can be compared in a likelihood ratio test.  Do you know of literature that shows that it is?  

Absent that, it may be very misleading to compare WLS models this way.  One argument that says it would be valid is that asymptotically WLS and ML should end up in the same place, but I'm not sure that the appeal to asymptotics is valid with finite sample sizes.  Maybe try a simulation to show that it's valid?

**Comment by tbates** [comment]
Hi mike. Hunter (above) described the implementation. Other apps implement and document the validity. But no, I can't implement this.

For context, Hunter implementation plan was:

1. Compute the independence model fit in the backend (1 fit function evaluation) and pass back as `attr(model@fitfunction$result, "IndependenceLikelihood")`

2. Evaluate fit function for `r'Wr = (s-sigma)'W(s-sigma)` either by zeroing the covariance elements of sigma and and setting everything else equal to the corresponding elements of s, OR by setting the non-covariance elements (means, thresholds, variances) of r to zero and leaving everything else alone.

**Comment by mcneale** [comment]
True, but just because other software has been doing pseudo-LRT doesn't mean that it's correct.  If it is, then it should happen for sure.
